client/blocks/stripe-utils/normalize.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @typedef {import('./type-defs').StripePaymentItem} StripePaymentItem
+ * @typedef {import('./type-defs').StripeShippingOption} StripeShippingOption
+ * @typedef {import('./type-defs').StripeShippingAddress} StripeShippingAddress
+ * @typedef {import('./type-defs').StripePaymentResponse} StripePaymentResponse
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').PreparedCartTotalItem} CartTotalItem
+ * @typedef {import('@woocommerce/type-defs/cart').CartShippingOption} CartShippingOption
+ * @typedef {import('@woocommerce/type-defs/shipping').ShippingAddress} CartShippingAddress
+ * @typedef {import('@woocommerce/type-defs/billing').BillingData} CartBillingAddress
+ */
+
+/**
+ * Normalizes incoming cart total items for use as a displayItems with the
+ * Stripe api.
+ *
+ * @param {CartTotalItem[]} cartTotalItems CartTotalItems to normalize
+ * @param {boolean}         pending        Whether to mark items as pending or
+ *                                         not
+ *
+ * @return {StripePaymentItem[]} An array of PaymentItems
+ */
+const normalizeLineItems = ( cartTotalItems, pending = false ) => {
+	return cartTotalItems
+		.map( ( cartTotalItem ) => {
+			return cartTotalItem.value
+				? {
+						amount: cartTotalItem.value,
+						label: cartTotalItem.label,
+						pending,
+				  }
+				: false;
+		} )
+		.filter( Boolean );
+};
+
+/**
+ * Normalizes incoming cart shipping option items for use as shipping options
+ * with the Stripe api.
+ *
+ * @param {CartShippingOption[]}  shippingOptions An array of CartShippingOption items.
+ *
+ * @return {StripeShippingOption[]}  An array of Stripe shipping option items.
+ */
+const normalizeShippingOptions = ( shippingOptions ) => {
+	const rates = shippingOptions[ 0 ].shipping_rates;
+	return rates.map( ( rate ) => {
+		return {
+			id: rate.rate_id,
+			label: rate.name,
+			detail: rate.description,
+			amount: parseInt( rate.price, 10 ),
+		};
+	} );
+};
+
+/**
+ * Normalize shipping address information from stripe's address object to
+ * the cart shipping address object shape.
+ *
+ * @param {StripeShippingAddress} shippingAddress Stripe's shipping address item
+ *
+ * @return {CartShippingAddress} The shipping address in the shape expected by
+ * the cart.
+ */
+const normalizeShippingAddressForCheckout = ( shippingAddress ) => {
+	const address = {
+		first_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 0, 1 )
+			.join( ' ' ),
+		last_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 1 )
+			.join( ' ' ),
+		company: '',
+		address_1:
+			typeof shippingAddress.addressLine[ 0 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 0 ],
+		address_2:
+			typeof shippingAddress.addressLine[ 1 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 1 ],
+		city: shippingAddress.city,
+		state: shippingAddress.region,
+		country: shippingAddress.country,
+		postcode: shippingAddress.postalCode.replace( ' ', '' ),
```

Comment:
All of these properties on `shippingAddress` [may be `undefined`](https://github.com/stripe/stripe-js/blob/db51d9576a66ff0de4326aeac19dc73934935bc8/types/stripe-js/payment-request.d.ts#L360-L382) so we need to check for their existence

```suggestion
		city: shippingAddress.city ?? '',
		state: shippingAddress.region ?? '',
		country: shippingAddress.country ?? '',
		postcode: shippingAddress.postalCode?.replace( ' ', '' ) ?? '',
```

Reply:
Fixed in f9473b13f23957b3e991ffafa14fc24e93d06b58